### PR TITLE
Add Char#+(str : String)

### PR DESCRIPTION
**Pull request by hangyas**
`Char#+(str : String)` concatenates a char and a string.
`Char#bytesize` returns the number of UTF-8 bytes.
```crystal
'f' + "oo" # => "foo"
'旅'.bytesize # => 3
```

### CODE REVIEW

File: src/char.cr [hunk 1]
```diff
@@ -51,18 +51,39 @@ struct Char
...
-  # 'c' - 'a' #=> 2
+  # 'a' - 'a' # => 0
+  # 'b' - 'a' # => 1
+  # 'c' - 'a' # => 2
   # ```
   def -(other : Char)
     ord - other.ord
   end
 
+  # Concatenates *char* and *string*.
```
Comment (jhass) [inline src/char.cr]: \*foo\* is doc convention to parameters, `char` is none.

**Comment by jhass** [comment]
`bytesize` might be a bit misleading, since `Char` is always a `UInt32` in memory (afaik). Maybe we can find a better name?

File: src/char.cr [hunk 2]
```diff
@@ -51,18 +51,39 @@ struct Char
...
+  #
+  # ```
+  # 'f' + "oo" # => "foo"
+  # ```
+  def +(str : String)
+    bytesize = str.bytesize + bytesize
+    s = String.new(bytesize) do |buffer|
+          count = 0
+          each_byte do |byte|
+            (buffer + count).copy_from(pointerof(byte), 1)
```
Comment (asterite) [inline src/char.cr]: `buffer[count] = byte` works too

**Comment by hangyas** [comment]
I was thinking about utf8size or something, but it already had each_byte which only yields the same number of bytes too

**Comment by asterite** [comment]
I actually think `bytesize` is OK. If a String by convention is UTF-8 encoded and its `bytesize` returns the number of bytes in that encoding, it feels natural for `Char#bytesize` to do the same. And the argument that takes into account `Char#each_byte` is also good and valid. So :+1: from me.

**Comment by sdogruyol** [comment]
:+1:

**Comment by jhass** [comment]
Alright, could you squash the commits please?

**Comment by jhass** [comment]
Merged as b60910d77f29c1d565ae40e0dff70cb1213783f3

Thanks!

**Pull request closed without merging**